components/global/portal/io.cellery.observability.ui/src/components/cells/cell/Details.js

```diff
@@ -79,43 +80,42 @@ class Details extends React.Component {
             includeIntraCell: true
         };
 
+        const ingressQueryParams = {
+            queryStartTime: queryStartTime.valueOf(),
+            queryEndTime: queryEndTime.valueOf()
+        };
+
+        const cellMetricsPromise = HttpUtils.callObservabilityAPI(
+            {
+                url: `/http-requests/cells/metrics/${HttpUtils.generateQueryParamString(search)}`,
+                method: "GET"
+            }, globalState);
+
+        const ingressDataPromise = HttpUtils.callObservabilityAPI(
+            {
+                url: `/k8s/cells/${cell}${HttpUtils.generateQueryParamString(ingressQueryParams)}`,
+                method: "GET"
+            }, this.props.globalState);
+
         if (isUserAction) {
             NotificationUtils.showLoadingOverlay("Loading Cell Info", globalState);
             self.setState({
                 isLoading: true
             });
         }
-        HttpUtils.callObservabilityAPI(
-            {
-                url: `/http-requests/cells/metrics/${HttpUtils.generateQueryParamString(search)}`,
-                method: "GET"
-            },
-            globalState
-        ).then((data) => {
-            const aggregatedData = data.map((datum) => ({
-                isError: datum[1] === "5xx",
-                count: datum[5]
-            })).reduce((accumulator, currentValue) => {
-                if (currentValue.isError) {
-                    accumulator.errorsCount += currentValue.count;
-                }
-                accumulator.total += currentValue.count;
-                return accumulator;
-            }, {
-                errorsCount: 0,
-                total: 0
-            });
-
-            let health;
-            if (aggregatedData.total > 0) {
-                health = 1 - aggregatedData.errorsCount / aggregatedData.total;
-            } else {
-                health = -1;
+        Promise.all([cellMetricsPromise, ingressDataPromise]).then((data) => {
+            const cellInfoData = data[1];
+            this.loadCellMetrics(data[0]);
+            const ingressTypeSet = new Set();
+            for (let i = 0; i < cellInfoData.length; i++) {
+                const ingressDatum = cellInfoData[i];
+                const ingressTypeArray = ingressDatum[2].split(",");
+                ingressTypeArray.forEach((ingressValue) => {
+                    ingressTypeSet.add(ingressValue);
+                });
             }
-
             self.setState({
-                health: health,
-                isDataAvailable: aggregatedData.total > 0
+                ingressTypes: Array.from(ingressTypeSet).join(", ")
```

Comment:
It's better to store the ingress types as an array in the state and build the string when it is required. Please fix in all other places as well